### PR TITLE
[xaprepare] Further steps to fix brew

### PR DESCRIPTION
**Pull request by grendello**
Turns out we can sometimes have a situation when a homebrew package is
installed but it's not linked into a directory that's found in the
user's `PATH` environment variable.

Remedy the situation by unconditionally linking all the packages we
depend on.

### CODE REVIEW

File: build-tools/xaprepare/xaprepare/ToolRunners/BrewRunner.MacOS.cs [hunk 1]
```diff
@@ -21,6 +21,8 @@ public BrewRunner (Context context, Log log = null, string toolPath = null)
 			: base (context, log, toolPath)
 		{
 			ProcessTimeout = TimeSpan.FromMinutes (30);
+			EchoStandardError = true;
```
Comment (jonpryor) [inline build-tools/xaprepare/xaprepare/ToolRunners/BrewRunner.MacOS.cs]: Not sure I see why these changes are needed?  I don' understand the interaction between these properties and the parameters to `Link()`.

File: build-tools/xaprepare/xaprepare/Application/HomebrewProgram.MacOS.cs [hunk 2]
```diff
@@ -190,6 +190,11 @@ protected override async Task AfterDetect (bool installed)
 				return;
 			}
 
+			// It may happen that the package is installed but not linked to the prefix that's in the user's PATH.
+			// Detecting whether the package is linked would require requesting and parsing JSON for all the packages which
+			// would be more trouble than it's worth. Let's just link the package
+			await runner.Link (Name, echoOutput: false, echoError: false);
```
Comment (jonpryor) [inline build-tools/xaprepare/xaprepare/Application/HomebrewProgram.MacOS.cs]: Why *wouldn't* we want to echo stdout & stderr?

File: build-tools/xaprepare/xaprepare/ToolRunners/BrewRunner.MacOS.cs (hunk 1 above)
Comment (grendello) [reply build-tools/xaprepare/xaprepare/ToolRunners/BrewRunner.MacOS.cs]: It's needed because without this we won't **capture** the output and each `xaprepare` run will be marred with the messages below for each dependency check:

```
Warning: Already linked: /usr/local/Cellar/mingw-w64/7.0.0_1
To relink:
  brew unlink mingw-w64 && brew link mingw-w64
```

File: build-tools/xaprepare/xaprepare/Application/HomebrewProgram.MacOS.cs (hunk 2 above)
Comment (grendello) [reply build-tools/xaprepare/xaprepare/Application/HomebrewProgram.MacOS.cs]: See my reply above. The `Warning` line is printed to `stderr` and would be shown in red, the remaining two lines are sent to `stdout`.
